Source/CrossingDetector/CrossingDetector.h

```diff
@@ -144,26 +158,21 @@ class CrossingDetector : public GenericProcessor
     Array<bool> futureBinary;
     //array to compare jumpLimit
     Array<float> jumpSize;
+    Array<float> thresholdHistory;
```

Comment:
I added a new array to save the threshold corresponding to each sample so that events can accurately save the threshold at the crossing in their metadata. The threshold may be changing frequently if random thresholds are enabled or (soon) if we're reading thresholds from a continuous input.